.github/workflows/main.yml

```diff
@@ -19,14 +19,15 @@ jobs:
     - uses: conda-incubator/setup-miniconda@v2
       with:
         auto-update-conda: false
-        miniconda-version: py37_4.10.3
         auto-activate-base: true
         activate-environment: ""
     - name: Install build dependencies
       shell: bash -l {0}
       run: |
         conda config --append channels conda-forge
-        conda install -y conda-build conda-verify flake8 pydocstyle yapf==0.30.0 sphinx sphinx_rtd_theme
+        # TODO: solver=libmamba is default for conda>=23.10 and thus can be removed for future Miniconda versions.
+        CONDA_SOLVER=libmamba \
+        conda install -y conda\>=23.10 conda-build conda-verify flake8 pydocstyle yapf==0.30.0 sphinx sphinx_rtd_theme
```

Comment:
Using an old `conda`/`python`
- caused installation of an incompatible `conda-build` version,
- is unnecessarily slow in conjunction with using both `defaults` and `conda-forge`.

Since this job is independent from the actual tests, this should be fine.
(N.B.: It'd be good if some one else could concern themselves with fixing the CI.)